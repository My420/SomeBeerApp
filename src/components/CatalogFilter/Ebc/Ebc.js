import React from 'react';
import styles from './Ebc.module.scss';
import {
  EBC_LESS_PROP,
  EBC_MORE_PROP,
  EBC_BOTTOM_VALUE,
  EBC_TOP_VALUE
} from '../../../utils/constants';
import validateNumberValue from '../../../utils/validateNumberValue';

class Ebc extends React.PureComponent {
  onEBCChange = evt => {
    const { onPropertyChange } = this.props;
    const value = validateNumberValue(
      evt.target.value,
      EBC_BOTTOM_VALUE,
      EBC_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'ebcLess') {
      onPropertyChange(value, EBC_LESS_PROP);
    } else if (direction === 'ebcMore') {
      onPropertyChange(value, EBC_MORE_PROP);
    }
  };

  render() {
    // eslint-disable-next-line no-console
    console.log('render ========== EbcInput');

    const { lessValue, moreValue } = this.props;
    return (
      <fieldset className={styles.ebcField}>
        <legend className={styles.ebclegend}>Color Units Ebc</legend>
        <label className={styles.ebcLabelMore} htmlFor="ebcMore">
          {'More'}
          <input
            id="ebcMore"
            name="ebcMore"
            className={styles.ebcInputMore}
            type="number"
            value={moreValue}
            onChange={this.onEBCChange}
          />
          <label className={styles.ebcLabelLess} htmlFor="ebcLess">
            {'Less'}
            <input
              id="ebcLess"
              name="ebcLess"
              className={styles.ebcInputLess}
              type="number"
              value={lessValue}
              onChange={this.onEBCChange}
            />
          </label>
        </label>
      </fieldset>
    );
  }
}

export default Ebc;
