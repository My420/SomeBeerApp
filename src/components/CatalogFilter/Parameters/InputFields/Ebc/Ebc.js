import React from 'react';
import PropTypes from 'prop-types';
import styles from './Ebc.module.scss';
import {
  EBC_LESS_PROP,
  EBC_MORE_PROP,
  EBC_BOTTOM_VALUE,
  EBC_TOP_VALUE
} from '../../../../../utils/constants';
import validateNumberValue from '../../../../../utils/validateNumberValue';

const Ebc = ({ lessValue, moreValue, onPropertyChange }) => {
  const onEBCChange = evt => {
    const value = validateNumberValue(
      evt.target.value,
      EBC_BOTTOM_VALUE,
      EBC_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'ebcLess') {
      onPropertyChange(value, EBC_LESS_PROP);
    } else if (direction === 'ebcMore') {
      onPropertyChange(value, EBC_MORE_PROP);
    }
  };

  return (
    <fieldset className={styles.ebcField}>
      <legend className={styles.ebcLegend}>Color Units Ebc</legend>
      <div className={styles.ebcInner}>
        <label className={styles.ebcLabelMore} htmlFor="ebcMore">
          <span className={styles.textMore}>more than</span>
          <input
            id="ebcMore"
            name="ebcMore"
            className={styles.ebcInputMore}
            type="number"
            value={moreValue}
            onChange={onEBCChange}
          />
        </label>
        <label className={styles.ebcLabelLess} htmlFor="ebcLess">
          <span className={styles.textLess}>less than</span>
          <input
            id="ebcLess"
            name="ebcLess"
            className={styles.ebcInputLess}
            type="number"
            value={lessValue}
            onChange={onEBCChange}
          />
        </label>
      </div>
    </fieldset>
  );
};

Ebc.propTypes = {
  lessValue: PropTypes.string.isRequired,
  moreValue: PropTypes.string.isRequired,
  onPropertyChange: PropTypes.func.isRequired
};

export default Ebc;
