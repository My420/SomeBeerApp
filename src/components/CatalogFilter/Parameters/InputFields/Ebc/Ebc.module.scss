@import '../../../../../sass/variables.scss';

.ebcField {
  padding-bottom: 10px;
  border: none;
}

.ebcInputMore,
.ebcInputLess {
  width: 50px;
  height: 25px;

  @include border;
  @include input;
}

.textMore,
.textLess {
  margin-right: 7px;
}

.ebcInner {
  display: flex;
  justify-content: space-around;
}

.ebcLegend {
  display: inline-block;
  text-align: center;
  font-weight: 700;
}

@media (max-width: 400px) {
  .ebcField {
    padding-bottom: 5px;
  }

  .ebcInputMore,
  .ebcInputLess {
    width: 30px;
    height: 20px;
    font-size: 14px;
  }
}
