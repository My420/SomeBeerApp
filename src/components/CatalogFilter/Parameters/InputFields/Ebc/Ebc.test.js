import React from 'react';
import { shallow } from 'enzyme';
import Ebc from './Ebc';

describe('test <Ebc /> component', () => {
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <Ebc lessValue="20" moreValue="30" onPropertyChange={mockFunc} />
  );

  beforeEach(() => {
    mockFunc.mockReset();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render  inputs with attr name='ebcLess' with correct value", () => {
    expect(wrapper.find("input[name='ebcLess']").prop('value')).toBe('20');
  });

  test("should render  inputs with attr name='ebcMore' with correct value", () => {
    expect(wrapper.find("input[name='ebcMore']").prop('value')).toBe('30');
  });

  test("should call onPropertyChange prop when inputs with attr name='ebcLess' change", () => {
    wrapper
      .find("input[name='ebcLess']")
      .simulate('change', { target: { value: '40', name: 'ebcLess' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('40');
  });

  test("should call onPropertyChange prop when inputs with attr name='ebcMore' change", () => {
    wrapper
      .find("input[name='ebcMore']")
      .simulate('change', { target: { value: '10', name: 'ebcMore' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('10');
  });
});
