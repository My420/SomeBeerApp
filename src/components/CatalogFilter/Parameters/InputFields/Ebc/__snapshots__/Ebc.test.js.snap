// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Ebc /> component should match snapshot 1`] = `
<fieldset
  className="ebcField"
>
  <legend
    className="ebcLegend"
  >
    Color Units Ebc
  </legend>
  <div
    className="ebcInner"
  >
    <label
      className="ebcLabelMore"
      htmlFor="ebcMore"
    >
      <span
        className="textMore"
      >
        more than
      </span>
      <input
        className="ebcInputMore"
        id="ebcMore"
        name="ebcMore"
        onChange={[Function]}
        type="number"
        value="30"
      />
    </label>
    <label
      className="ebcLabelLess"
      htmlFor="ebcLess"
    >
      <span
        className="textLess"
      >
        less than
      </span>
      <input
        className="ebcInputLess"
        id="ebcLess"
        name="ebcLess"
        onChange={[Function]}
        type="number"
        value="20"
      />
    </label>
  </div>
</fieldset>
`;
